Add use of Configuration, Swagger2 and Scheduled Jobs

santie88 opened the pull request.

santie88 committed:
Add AppConfig and Conditional Beans

santie88 committed:
Add of Swagger2 and Scheduled Job

juanalmeydagoeuro commented on src/main/java/com/example/springbootrestexample/configurations/AppConfig.java:
```diff
@@ -0,0 +1,46 @@
+@Configuration
+public class AppConfig {
+
+    @Bean(name="bookRepository")
```
It is not a good practice to put names to the beans. It is enough information with the type (interface)

juanalmeydagoeuro commented on src/main/java/com/example/springbootrestexample/logic/BookLogic.java:
```diff
@@ -16,8 +18,10 @@
 public class BookLogic extends ExceptionMessageLogic {
 
     @Autowired
+    @Qualifier("bookRepository")
```
Remove qualifiers

juanalmeydagoeuro replied:
and autowires

juanalmeydagoeuro replied:
Dependencies should be added in the constructor, so the colaborators should have the modifiers
private final...

juanalmeydagoeuro commented on src/main/java/com/example/springbootrestexample/scheduledJobs/BookChecking.java:
```diff
@@ -0,0 +1,23 @@
+    @Scheduled(fixedDelayString = "${schedule.frequency}")
+    public void CheckBook(){
+        Book book = bookLogic.getBookByIsbn("1");
+        book.setLastCheck(Calendar.getInstance().getTime());
```
Use DateTime (the library that came with Java 8)

juanalmeydagoeuro commented on src/main/java/com/example/springbootrestexample/configurations/AppConfig.java:
```diff
@@ -0,0 +1,46 @@
+    }
+
+    @Bean(name="bookRepository")
+    @Conditional(MemoryConfiguration.class)
```
Do not base the configurations in the @Conditional base those in the spring environment. It can be configurated in @Configuration and set in the program arguments:  -Dspring.profiles.active=
I would recommend to read: https://docs.spring.io/spring-boot/docs/current/reference/html/boot-features-profiles.html
In this case I would use in memory persistence for dev, and mongo db for prod

juanalmeydagoeuro commented on src/main/java/com/example/springbootrestexample/configurations/MemoryConfiguration.java:
```diff
@@ -0,0 +1,13 @@
+import org.springframework.context.annotation.ConditionContext;
+import org.springframework.core.type.AnnotatedTypeMetadata;
+
+public class MemoryConfiguration implements Condition {
```
Nice implementation but it could be done better with profiles. Check previous comment.

santie88 committed:
Correction of PullRequest of branch: Add use of Configuration, Swagger2 and Scheduled Jobs
1. Change use of Conditional for Profile
2. Repository Entities, do not use Autowired or Qualifier anymore.
3. Change of Book.LastChecked to LocalDateTime.
